c10/util/typeid.h

```diff
@@ -152,6 +171,7 @@ struct TypeMetaData final {
   Delete* delete_;
   TypeIdentifier id_;
   c10::string_view name_;
+  c10::optional<ScalarType> scalarTypeOpt_;
```

Comment:
technically you don't need this one and could resort to just the pointer arithmetic (it's faster!)

Reply:
See #44965 😁